src/flake8_simplify/plugins/unary_ops.rs

```diff
@@ -0,0 +1,127 @@
+use rustpython_ast::{Cmpop, Expr, ExprKind, Location, StmtKind, Unaryop};
+
+use crate::ast::types::Range;
+use crate::autofix::Fix;
+use crate::checkers::ast::Checker;
+use crate::pycodestyle::plugins::compare;
+use crate::registry::{Check, CheckKind};
+use crate::source_code_generator::SourceCodeGenerator;
+
+fn is_exception_check(stmt: &StmtKind) -> bool {
+    let StmtKind::If {test: _, body, orelse: _} = stmt else {
+        return false;
+    };
+    if body.len() != 1 {
+        return false;
+    }
+    if matches!(body[0].node, StmtKind::Raise { .. }) {
+        return true;
+    }
+    false
+}
+
+/// SIM201
+pub fn negation_with_equal_op(checker: &mut Checker, expr: &Expr, op: &Unaryop, operand: &Expr) {
+    if !matches!(op, Unaryop::Not) {
+        return;
+    }
+    let ExprKind::Compare{ left, ops, comparators} = &operand.node else {
+        return;
+    };
+    if !matches!(&ops[..], [Cmpop::Eq]) {
+        return;
+    }
+    let parent = &checker.current_stmt().0.node;
+    if is_exception_check(parent) {
+        return;
+    }
+    let mut check = Check::new(
+        CheckKind::NegateEqualOp(left.to_string(), comparators[0].to_string()),
```

Comment:
I know it's a little tedious but we should avoid calling `.to_string()` on expressions and statements.

Instead we should _either_ grab them with `SourceCodeLocator` (`checker.locator`) or round-trip them with `SourceCodeGenerator` to ensure that they're in the proper style. I've been preferring the latter for the SIM checks. Like:

```rs
let mut generator = SourceCodeGenerator::new(
    checker.style.indentation(),
    checker.style.quote(),
    checker.style.line_ending(),
);
generator.unparse_expr(&left, 0);
generator.generate()
```

Reply:
Thanks, ended up moving above to `fn expr_with_style` and use it when generating `String`s for `Check` and `Fix`.